src/engine/ResultTable.cpp

```diff
@@ -11,7 +11,8 @@ ResultTable::ResultTable()
       _sortedBy(0),
       _varSizeData(),
       _fixedSizeData(nullptr),
-      _status(ResultTable::OTHER) {}
+      _status(ResultTable::OTHER),
+      _localVocab(std::make_shared<std::vector<std::string>>()) {}
```

Comment:
What's the advantage of having this as a `shared_ptr`?

Reply:
It prevents copying of the contents of _localVocab after evaluating a subtree. As elements in the vocabulary are currently never moved or deleted this should not create any problems.

Reply:
Ok, I think this warrants a comment. Also can't we just `std::move()` the `_localVocab`?

Reply:
I think we could use std::move, the only advantage of using shared_pointers that I can think of right now is that the old ResultTable does not loose access to the vocabulary after it was copied. That would probably not be a problem right now, but cost wise shared pointers should be very close to std::move, and they avoid problems with interpreting the old results after already having moved the vocabulary.